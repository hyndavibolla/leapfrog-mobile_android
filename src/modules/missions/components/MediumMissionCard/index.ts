export { default as MediumMissionCard } from './MediumMissionCard';
