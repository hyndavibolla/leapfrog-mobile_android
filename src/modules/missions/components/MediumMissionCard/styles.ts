import { css } from '@emotion/native';

import { COLOR, FONT_SIZE } from '_constants';

export const styles = {
  cardContainer: css`
    background-color: ${COLOR.WHITE};
    border-radius: 8px;
    width: 160px;
    height: 200px;
    justify-content: flex-start;
    padding: 0;
    margin: 8px;
    overflow: hidden;
  `,
  backgroundContainer: css`
    border-top-left-radius: 8px;
    border-top-right-radius: 8px;
    height: 80px;
    background-color: ${COLOR.GRAY};
    overflow: hidden;
  `,
  logoContainer: css`
    position: absolute;
    align-self: center;
    margin-top: 45px;
    border: 5px solid ${COLOR.WHITE};
    border-radius: 60px;
  `,
  logoImage: css`
    width: 60px;
    height: 60px;
    border-radius: 60px;
    background-color: ${COLOR.DARK_GRAY};
  `,
  title: css`
    color: ${COLOR.BLACK};
    font-size: ${FONT_SIZE.SMALL};
    text-align: center;
    justify-content: center;
  `,
  infoContainer: css`
    padding: 40px 10px 10px 10px;
    flex: 1;
    align-items: center;
    justify-content: space-around;
  `,
  pill: css`
    align-self: center;
  `,
  missionOfferContainer: css`
    background-color: ${COLOR.PRIMARY_BLUE};
    width: 100%;
    height: 24px;
    position: absolute;
    top: 0;
    left: 0;
  `,
  missionOfferText: css`
    font-size: ${FONT_SIZE.TINY};
    color: ${COLOR.WHITE};
    text-align: center;
  `
};
