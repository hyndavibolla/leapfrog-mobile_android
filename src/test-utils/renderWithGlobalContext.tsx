import React, { ReactElement } from 'react';
import { render } from '@testing-library/react-native';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWithGlobalContext = (
  children: ReactElement,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const renderAPI = render(
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderAPI, mockReducer: combinedReducers, deps };
};
