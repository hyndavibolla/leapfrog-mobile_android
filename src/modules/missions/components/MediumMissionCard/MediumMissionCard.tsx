import React, { memo } from 'react';
import { View, TouchableHighlight } from 'react-native';

import { FONT_FAMILY } from '_constants';
import { BrandLogo } from '_components/BrandLogo';
import { Card } from '_components/Card';
import { ImageBackgroundWithFallback } from '_components/ImageBackgroundWithFallback';
import { Pill } from '_components/Pill';
import { Text } from '_components/Text';
import { MissionModel } from '_models';
import { getMissionPointsAwardedText } from '_utils/getMissionPointsAwardedText';
import { useTestingHelper } from '_utils/useTestingHelper';
import { styles } from './styles';

export interface Props {
  mission: MissionModel.IMission;
  hasStreakTag?: boolean;
  onPress?: () => void;
}

export const MediumMissionCard = ({ onPress, mission, hasStreakTag }: Props) => {
  const { getTestIdProps } = useTestingHelper('medium-mission-card');
  return (
    <TouchableHighlight underlayColor="transparent" onPress={onPress} {...getTestIdProps('container')}>
      <Card style={styles.cardContainer}>
        <ImageBackgroundWithFallback
          source={{ uri: mission.image }}
          containerStyle={styles.backgroundContainer}
          fallbackSource={require('_assets/shared/missionMediumCardBg.png')}
        />
        <View style={styles.logoContainer}>
          <BrandLogo image={mission?.brandLogo} category={mission?.pointsAwarded?.conditions[0]?.category} style={styles.logoImage} />
        </View>
        <View style={styles.infoContainer}>
          <Text font={FONT_FAMILY.BOLD} numberOfLines={1} ellipsizeMode="tail" style={styles.title} {...getTestIdProps('mission-name')}>
            {mission.brandName}
          </Text>
          <Pill style={styles.pill} textFallback={'Mission'}>
            {getMissionPointsAwardedText(mission.pointsAwarded)}
          </Pill>
        </View>
        {hasStreakTag && (
          <View style={styles.missionOfferContainer} {...getTestIdProps('streak-tag')}>
            <Text style={styles.missionOfferText}>ðŸ”¥ Mission offer</Text>
          </View>
        )}
      </Card>
    </TouchableHighlight>
  );
};

export default memo(MediumMissionCard);
