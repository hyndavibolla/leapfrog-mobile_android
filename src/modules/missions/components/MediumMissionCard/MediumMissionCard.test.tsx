import React from 'react';
import { fireEvent } from '@testing-library/react-native';

import { getMission_1, getMission_2, getMission_3 } from '_test_utils/entities';
import { renderWithGlobalContext } from '_test_utils/renderWithGlobalContext';
import MediumMissionCard, { Props } from './MediumMissionCard';

describe('MediumMissionCard', () => {
  let onPress: () => void;
  let props: Props;
  beforeEach(() => {
    onPress = jest.fn();
    props = {
      onPress,
      mission: getMission_1()
    };
  });

  it('should render', () => {
    const { toJSON } = renderWithGlobalContext(<MediumMissionCard {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render with points and dollars mission', () => {
    const { toJSON } = renderWithGlobalContext(<MediumMissionCard {...props} mission={getMission_2()} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render with points only mission', () => {
    const { toJSON } = renderWithGlobalContext(<MediumMissionCard {...props} mission={getMission_3()} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should be pressable ', () => {
    const { getByTestId } = renderWithGlobalContext(<MediumMissionCard {...props} />);
    fireEvent.press(getByTestId('medium-mission-card-container'));
    expect(props.onPress).toBeCalled();
  });

  it('should render with streak tag', () => {
    props.hasStreakTag = true;
    const { getByTestId } = renderWithGlobalContext(<MediumMissionCard {...props} />);
    expect(getByTestId('medium-mission-card-streak-tag')).toBeTruthy();
  });
});
