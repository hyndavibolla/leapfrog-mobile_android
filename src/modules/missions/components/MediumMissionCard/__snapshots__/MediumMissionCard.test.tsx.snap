// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediumMissionCard should render 1`] = `
<View
  accessibilityLabel="medium-mission-card-container"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="medium-mission-card-container"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "elevation": 3,
          "shadowColor": "#262C2F",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.1,
        },
        Object {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "height": 200,
          "justifyContent": "flex-start",
          "marginBottom": 8,
          "marginLeft": 8,
          "marginRight": 8,
          "marginTop": 8,
          "overflow": "hidden",
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "width": 160,
        },
      ]
    }
  >
    <View
      accessibilityLabel="image-background-container"
      accessible={true}
      style={
        Object {
          "backgroundColor": "#E2E3E3",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "height": 80,
          "overflow": "hidden",
        }
      }
      testID="image-background-container"
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 80,
                "width": undefined,
              },
              undefined,
            ],
          ]
        }
      >
        <FastImageView
          accessibilityLabel="image-background-image"
          accessible={true}
          onFastImageError={[Function]}
          onFastImageLoadEnd={[Function]}
          onFastImageLoadStart={[Function]}
          resizeMode="cover"
          source={
            Object {
              "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTVkMt3KFinkf_ehnCq5yk2DLZBls6GYFZLeA&usqp=CAU",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          testID="image-background-image"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "borderBottomLeftRadius": 60,
          "borderBottomRightRadius": 60,
          "borderColor": "#FFFFFF",
          "borderStyle": "solid",
          "borderTopLeftRadius": 60,
          "borderTopRightRadius": 60,
          "borderWidth": 5,
          "marginTop": 45,
          "position": "absolute",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 60,
              "width": 60,
            },
            Object {
              "backgroundColor": "#909BA0",
              "borderBottomLeftRadius": 60,
              "borderBottomRightRadius": 60,
              "borderTopLeftRadius": 60,
              "borderTopRightRadius": 60,
              "height": 60,
              "width": 60,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "alignItems": "center",
                "borderBottomLeftRadius": 60,
                "borderBottomRightRadius": 60,
                "borderTopLeftRadius": 60,
                "borderTopRightRadius": 60,
                "height": "100%",
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <FastImageView
            accessibilityLabel="brand-logo-image"
            accessible={true}
            onFastImageError={[Function]}
            onFastImageLoad={[Function]}
            onFastImageLoadEnd={[Function]}
            onFastImageLoadStart={[Function]}
            resizeMode="cover"
            source={
              Object {
                "priority": "high",
                "uri": "https://www.movistar.es/estaticos/img/ico-app-mi-movistar-2017.png",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID="brand-logo-image"
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-around",
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 40,
        }
      }
    >
      <Text
        accessibilityLabel="medium-mission-card-mission-name"
        accessible={true}
        ellipsizeMode="tail"
        font="SFProDisplay-Bold"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": 16,
              "lineHeight": 22,
            },
            Object {
              "color": "#262C2F",
              "fontSize": 18,
              "justifyContent": "center",
              "textAlign": "center",
            },
          ]
        }
        testID="medium-mission-card-mission-name"
      >
        Starbucks
      </Text>
      <View
        accessibilityLabel="pill-container"
        accessible={true}
        style={
          Array [
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#DDE8F5",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "paddingLeft": 4,
                "paddingRight": 10,
              },
              Object {
                "height": 28,
              },
            ],
            Array [
              undefined,
            ],
            Object {
              "alignSelf": "center",
            },
          ]
        }
        testID="pill-container"
      >
        <View
          accessibilityLabel="pill-icon"
          accessible={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderBottomLeftRadius": 50,
                "borderBottomRightRadius": 50,
                "borderTopLeftRadius": 50,
                "borderTopRightRadius": 50,
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          testID="pill-icon"
        >
          <Text
            accessibilityLabel="icon-text"
            accessible={true}
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#0066CC",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "sywmax",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="icon-text"
          >
            î¥‰
          </Text>
        </View>
        <Text
          accessibilityLabel="pill-text"
          accessible={true}
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Array [
                Object {
                  "paddingLeft": 4,
                },
                Array [
                  Object {
                    "color": "#0066CC",
                    "fontFamily": "SFProDisplay-Bold",
                    "fontSize": 14,
                  },
                  Object {
                    "lineHeight": 20,
                  },
                  undefined,
                ],
              ],
            ]
          }
          testID="pill-text"
        >
          200
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`MediumMissionCard should render with points and dollars mission 1`] = `
<View
  accessibilityLabel="medium-mission-card-container"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="medium-mission-card-container"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "elevation": 3,
          "shadowColor": "#262C2F",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.1,
        },
        Object {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "height": 200,
          "justifyContent": "flex-start",
          "marginBottom": 8,
          "marginLeft": 8,
          "marginRight": 8,
          "marginTop": 8,
          "overflow": "hidden",
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "width": 160,
        },
      ]
    }
  >
    <View
      accessibilityLabel="image-background-container"
      accessible={true}
      style={
        Object {
          "backgroundColor": "#E2E3E3",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "height": 80,
          "overflow": "hidden",
        }
      }
      testID="image-background-container"
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 80,
                "width": undefined,
              },
              undefined,
            ],
          ]
        }
      >
        <FastImageView
          accessibilityLabel="image-background-image"
          accessible={true}
          onFastImageError={[Function]}
          onFastImageLoadEnd={[Function]}
          onFastImageLoadStart={[Function]}
          resizeMode="cover"
          source={
            Object {
              "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTVkMt3KFinkf_ehnCq5yk2DLZBls6GYFZLeA&usqp=CAU",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          testID="image-background-image"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "borderBottomLeftRadius": 60,
          "borderBottomRightRadius": 60,
          "borderColor": "#FFFFFF",
          "borderStyle": "solid",
          "borderTopLeftRadius": 60,
          "borderTopRightRadius": 60,
          "borderWidth": 5,
          "marginTop": 45,
          "position": "absolute",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 60,
              "width": 60,
            },
            Object {
              "backgroundColor": "#909BA0",
              "borderBottomLeftRadius": 60,
              "borderBottomRightRadius": 60,
              "borderTopLeftRadius": 60,
              "borderTopRightRadius": 60,
              "height": 60,
              "width": 60,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "alignItems": "center",
                "borderBottomLeftRadius": 60,
                "borderBottomRightRadius": 60,
                "borderTopLeftRadius": 60,
                "borderTopRightRadius": 60,
                "height": "100%",
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <FastImageView
            accessibilityLabel="brand-logo-image"
            accessible={true}
            onFastImageError={[Function]}
            onFastImageLoad={[Function]}
            onFastImageLoadEnd={[Function]}
            onFastImageLoadStart={[Function]}
            resizeMode="cover"
            source={
              Object {
                "priority": "high",
                "uri": "https://www.vodafone.es/c/particulares/es/acceso-area-privada/mvf/static/img/modules/backgrounds/New_VF_Icon_RGB_RED.png",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID="brand-logo-image"
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-around",
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 40,
        }
      }
    >
      <Text
        accessibilityLabel="medium-mission-card-mission-name"
        accessible={true}
        ellipsizeMode="tail"
        font="SFProDisplay-Bold"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": 16,
              "lineHeight": 22,
            },
            Object {
              "color": "#262C2F",
              "fontSize": 18,
              "justifyContent": "center",
              "textAlign": "center",
            },
          ]
        }
        testID="medium-mission-card-mission-name"
      >
        Walmart
      </Text>
      <View
        accessibilityLabel="pill-container"
        accessible={true}
        style={
          Array [
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#DDE8F5",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "paddingLeft": 4,
                "paddingRight": 10,
              },
              Object {
                "height": 28,
              },
            ],
            Array [
              undefined,
            ],
            Object {
              "alignSelf": "center",
            },
          ]
        }
        testID="pill-container"
      >
        <View
          accessibilityLabel="pill-icon"
          accessible={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderBottomLeftRadius": 50,
                "borderBottomRightRadius": 50,
                "borderTopLeftRadius": 50,
                "borderTopRightRadius": 50,
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          testID="pill-icon"
        >
          <Text
            accessibilityLabel="icon-text"
            accessible={true}
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#0066CC",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "sywmax",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="icon-text"
          >
            î¥‰
          </Text>
        </View>
        <Text
          accessibilityLabel="pill-text"
          accessible={true}
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Array [
                Object {
                  "paddingLeft": 4,
                },
                Array [
                  Object {
                    "color": "#0066CC",
                    "fontFamily": "SFProDisplay-Bold",
                    "fontSize": 14,
                  },
                  Object {
                    "lineHeight": 20,
                  },
                  undefined,
                ],
              ],
            ]
          }
          testID="pill-text"
        >
          100 per $1
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`MediumMissionCard should render with points only mission 1`] = `
<View
  accessibilityLabel="medium-mission-card-container"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="medium-mission-card-container"
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "elevation": 3,
          "shadowColor": "#262C2F",
          "shadowOffset": Object {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.1,
        },
        Object {
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "height": 200,
          "justifyContent": "flex-start",
          "marginBottom": 8,
          "marginLeft": 8,
          "marginRight": 8,
          "marginTop": 8,
          "overflow": "hidden",
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "width": 160,
        },
      ]
    }
  >
    <View
      accessibilityLabel="image-background-container"
      accessible={true}
      style={
        Object {
          "backgroundColor": "#E2E3E3",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "height": 80,
          "overflow": "hidden",
        }
      }
      testID="image-background-container"
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 80,
                "width": undefined,
              },
              undefined,
            ],
          ]
        }
      >
        <FastImageView
          accessibilityLabel="image-background-image"
          accessible={true}
          onFastImageError={[Function]}
          onFastImageLoadEnd={[Function]}
          onFastImageLoadStart={[Function]}
          resizeMode="cover"
          source={
            Object {
              "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTVkMt3KFinkf_ehnCq5yk2DLZBls6GYFZLeA&usqp=CAU",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          testID="image-background-image"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "borderBottomLeftRadius": 60,
          "borderBottomRightRadius": 60,
          "borderColor": "#FFFFFF",
          "borderStyle": "solid",
          "borderTopLeftRadius": 60,
          "borderTopRightRadius": 60,
          "borderWidth": 5,
          "marginTop": 45,
          "position": "absolute",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 60,
              "width": 60,
            },
            Object {
              "backgroundColor": "#909BA0",
              "borderBottomLeftRadius": 60,
              "borderBottomRightRadius": 60,
              "borderTopLeftRadius": 60,
              "borderTopRightRadius": 60,
              "height": 60,
              "width": 60,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "alignItems": "center",
                "borderBottomLeftRadius": 60,
                "borderBottomRightRadius": 60,
                "borderTopLeftRadius": 60,
                "borderTopRightRadius": 60,
                "height": "100%",
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <FastImageView
            accessibilityLabel="brand-logo-image"
            accessible={true}
            onFastImageError={[Function]}
            onFastImageLoad={[Function]}
            onFastImageLoadEnd={[Function]}
            onFastImageLoadStart={[Function]}
            resizeMode="cover"
            source={
              Object {
                "priority": "high",
                "uri": "https://i.ibb.co/nMJXNWq/Logos-SYW-MAX.png",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID="brand-logo-image"
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-around",
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 40,
        }
      }
    >
      <Text
        accessibilityLabel="medium-mission-card-mission-name"
        accessible={true}
        ellipsizeMode="tail"
        font="SFProDisplay-Bold"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": 16,
              "lineHeight": 22,
            },
            Object {
              "color": "#262C2F",
              "fontSize": 18,
              "justifyContent": "center",
              "textAlign": "center",
            },
          ]
        }
        testID="medium-mission-card-mission-name"
      >
        Walmart
      </Text>
      <View
        accessibilityLabel="pill-container"
        accessible={true}
        style={
          Array [
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#DDE8F5",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "paddingLeft": 4,
                "paddingRight": 10,
              },
              Object {
                "height": 28,
              },
            ],
            Array [
              undefined,
            ],
            Object {
              "alignSelf": "center",
            },
          ]
        }
        testID="pill-container"
      >
        <View
          accessibilityLabel="pill-icon"
          accessible={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderBottomLeftRadius": 50,
                "borderBottomRightRadius": 50,
                "borderTopLeftRadius": 50,
                "borderTopRightRadius": 50,
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          testID="pill-icon"
        >
          <Text
            accessibilityLabel="icon-text"
            accessible={true}
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#0066CC",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "sywmax",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="icon-text"
          >
            î¥‰
          </Text>
        </View>
        <Text
          accessibilityLabel="pill-text"
          accessible={true}
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Array [
                Object {
                  "paddingLeft": 4,
                },
                Array [
                  Object {
                    "color": "#0066CC",
                    "fontFamily": "SFProDisplay-Bold",
                    "fontSize": 14,
                  },
                  Object {
                    "lineHeight": 20,
                  },
                  undefined,
                ],
              ],
            ]
          }
          testID="pill-text"
        >
          200
        </Text>
      </View>
    </View>
  </View>
</View>
`;
